api/api-internal-pipeline-groups-v1/src/main/java/com/thoughtworks/go/apiv1/internalpipelinegroups/representers/InternalPipelineGroupsRepresenter.java

```diff
@@ -0,0 +1,73 @@
+import com.thoughtworks.go.config.EnvironmentConfig;
+import com.thoughtworks.go.config.PipelineConfig;
+import com.thoughtworks.go.config.remote.ConfigOrigin;
+import com.thoughtworks.go.config.remote.FileConfigOrigin;
+import com.thoughtworks.go.config.remote.RepoConfigOrigin;
+
+import java.util.Collection;
+
+public class InternalPipelineGroupsRepresenter {
+    public static void toJSON(OutputWriter outputWriter, PipelineGroupsViewModel pipelineGroupsViewModel) {
+        outputWriter.
+                addChildList("groups", groupsWriter -> {
+                    pipelineGroupsViewModel.getPipelineGroups().forEach(group -> {
+                        groupsWriter.addChild(groupWriter -> {
+                            groupWriter.add("name", group.getGroup())
+                                    .addChildList("pipelines",
+                                            outputListWriter -> group.forEach(pipelineConfig -> {
+                                                outputListWriter.addChild(pipelineWriter -> {
+                                                    pipelineWriter.add("name", pipelineConfig.name());
+                                                    writeOrigin(pipelineWriter, pipelineConfig.getOrigin());
+                                                    renderEnvironment(pipelineWriter, pipelineConfig, pipelineGroupsViewModel);
+                                                });
+                                            }));
+                        });
+                    });
+                });
+    }
+
+    private static void renderEnvironment(OutputWriter pipelineWriter, PipelineConfig pipelineConfig, PipelineGroupsViewModel environments) {
+        EnvironmentConfig envForPipeline = environments.environmentFor(pipelineConfig.name());
+        if (envForPipeline != null) {
+            pipelineWriter.add("environment", envForPipeline.name());
+        } else {
+            pipelineWriter.renderNull("environment");
+        }
+    }
+
+    private static void writeOrigin(OutputWriter jsonWriter, ConfigOrigin origin) {
+        if (origin instanceof FileConfigOrigin) {
+            jsonWriter.addChild("origin", originWriter -> originWriter.add("type", "gocd"));
+        } else if (origin instanceof RepoConfigOrigin) {
+            jsonWriter.addChild("origin", originWriter -> {
+                originWriter.add("type", "config_repo");
+                originWriter.add("id", ((RepoConfigOrigin) origin).getConfigRepo().getId());
+            });
+        }
+    }
+
+    public static void toJSON(OutputWriter outputWriter, PipelineGroupsViewModel pipelineGroupsViewModel, Collection<String> users, Collection<String> roles) {
+        toJSON(outputWriter, pipelineGroupsViewModel);
+        outputWriter.addChild("additional_info", (writer) -> writer.addChildList("users", users)
+                .addChildList("roles", roles));
+    }
```

Comment:
This is not required as the envs page does not need the users and roles info.

Reply:
Done